target/linux/ramips/image/Makefile

```diff
@@ -56,6 +57,22 @@ define Build/loader-kernel
 	$(call Build/loader-common,LOADER_DATA="$@")
 endef
 
+define Build/loader-common-sercomm
+	rm -rf $@.src
+	$(MAKE) -C lzma-loader \
+		PKG_BUILD_DIR="$@.src" \
+		TARGET_DIR="$(dir $@)" LOADER_NAME="$(notdir $@)" \
+		BOARD="$(BOARDNAME)" PLATFORM="$(LOADER_PLATFORM)" \
+		LZMA_TEXT_START=0x82800000 LOADADDR=0x80001000 \
```

Comment:
What's wrong with using the variable with LOADADDR ...?

Reply:
I'll try to explain. We need LOADADDR=0x80001000 here and KERNEL_LOADADDR := 0x81001000 in all other places. Otherwise the devices hangs on boot. I tried to do something but on a not elegant way I think. For example few strings are longer than 73 chars. So I need your advice.

Reply:
KERNEL_LOADADDR is actually a device-variable, at least it's defined that way.

Thus, you could simply put "KERNEL_LOADADDR := 0x81001000" into the _device_ definition block, where it would be overwritten just for this device, and leave it untouched for all others.

Reply:
Ok, I found another yet option - in current commit.  

The essense if the problem: is if we simple put "KERNEL_LOADADDR := 0x81001000" into the device definition block then this value is used in all device recipes below because they have no own KERNEL_LOADADDR :=  & LZMA_TEXT_START := assignment. I tested that. Therefore KERNEL_LOADADDR :=  & LZMA_TEXT_START := should be assigned in every device recipe that use Build/loader-kernel.

So I made the following changes:
1. Defined device variables KERNEL_LOADADDR & LZMA_TEXT_START;
2. They are assigned for every device via Device/Default recipe and might be overriden in the device recipe.

It works, we have correct load addressess (and lzma text starts too). Tested on this device and Xiaomi AC2100:
binwalk --signature openwrt-ramips-mt7621-beeline_smartbox-giga-squashfs-kernel0.bin
```
DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
256           0x100           uImage header, header size: 64 bytes, header CRC: 0xB3629FFA, created: 2021-07-03 11:55:58, image size: 2518229 bytes, Data Address: 0x81001000, Entry Point: 0x81001000, data CRC: 0xD2D58AD2, OS: Linux, CPU: MIPS, image type: OS Kernel Image, compression type: lzma, image name: "MIPS OpenWrt Linux-5.4.128"
320           0x140           LZMA compressed data, properties: 0x6D, dictionary size: 8388608 bytes, uncompressed size: 2486881 bytes
```
binwalk --signature openwrt-ramips-mt7621-xiaomi_redmi-router-ac2100-squashfs-kernel1.bin
```
DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
0             0x0             uImage header, header size: 64 bytes, header CRC: 0x7DF37C72, created: 2021-07-03 11:55:58, image size: 2486868 bytes, Data Address: 0x80001000, Entry Point: 0x80001000, data CRC: 0xD86D046F, OS: Linux, CPU: MIPS, image type: OS Kernel Image, compression type: none, image name: "MIPS OpenWrt Linux-5.4.128"
5484          0x156C          Copyright string: "Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>"
5692          0x163C          LZMA compressed data, properties: 0x6D, dictionary size: 8388608 bytes, uncompressed size: 7893429 bytes
```

Reply:
@adschm Should I fix anything else?